### PR TITLE
World-builder - Skyboxes - load from BE, new component

### PR DESCRIPTION
**Pull request by dmitry-yudakov**
… previewImage

### CODE REVIEW

File: packages/app/src/api/repositories/assets3dRepository/assets3dRepository.api.endpoints.ts [hunk 1]
```diff
@@ -4,6 +4,7 @@ export const assets3dRepositoryEndpoints = () => {
   const BASE_URL = `${appVariables.BACKEND_API_URL}/assets-3d`;
 
   return {
-    upload: `${BASE_URL}/upload`
+    upload: `${BASE_URL}/upload`,
+    get: BASE_URL
```
Comment (Daboo1998) [inline packages/app/src/api/repositories/assets3dRepository/assets3dRepository.api.endpoints.ts]: Base or root, cause what if in the future there will be a post for same EP?

File: packages/app/src/scenes/worldBuilder/pages/WorldBuilderCustomizePanel/components/Carousel/Carousel.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  renderItem: (item: T, index: number) => JSX.Element;
+}
+
+export const Carousel = <T extends object>({
+  items,
+  activeItem,
+  windowSize = 5,
+  onChange,
+  renderItem
+}: PropsInterface<T>): JSX.Element => {
```
Comment (Daboo1998) [inline packages/app/src/scenes/worldBuilder/pages/WorldBuilderCustomizePanel/components/Carousel/Carousel.tsx:19]: Why not use FC?

File: packages/app/src/scenes/worldBuilder/pages/WorldBuilderCustomizePanel/components/Carousel/Carousel.tsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  const realWindowSize = Math.min(windowSize, items.length);
+  const halfSize = Math.floor(realWindowSize / 2);
+
+  const longItems = [...items, ...items, ...items];
```
Comment (Daboo1998) [inline packages/app/src/scenes/worldBuilder/pages/WorldBuilderCustomizePanel/components/Carousel/Carousel.tsx:21]: What is that for? Why three times the spread?

File: packages/app/src/scenes/worldBuilder/stores/WorldBuilderSkyboxesStore/WorldBuilderSkyboxesStore.ts [hunk 4]
```diff
@@ -34,7 +18,27 @@ const WorldBuilderSkyboxesStore = types
...
   .actions((self) => ({
     fetchItems: flow(function* () {
-      // TODO
+      const assets3d: FetchAssets3dResponse = yield self.request.send<null, FetchAssets3dResponse>(
+        api.assets3dRepository.fetchAssets3d,
+        null
+      );
+      console.log('Assets3d response:', assets3d);
+      if (!assets3d) {
+        console.log('Error loading assets3d');
```
Comment (Daboo1998) [inline packages/app/src/scenes/worldBuilder/stores/WorldBuilderSkyboxesStore/WorldBuilderSkyboxesStore.ts]: Consider using console.error

**Review by Daboo1998 — CHANGES_REQUESTED** [review]

File: packages/app/src/scenes/worldBuilder/pages/WorldBuilderCustomizePanel/components/Carousel/Carousel.tsx (hunk 3 above)
Comment (dmitry-yudakov) [reply packages/app/src/scenes/worldBuilder/pages/WorldBuilderCustomizePanel/components/Carousel/Carousel.tsx:21]: To let them circle and always have the currently selected element in the middle

File: packages/app/src/scenes/worldBuilder/pages/WorldBuilderCustomizePanel/components/Carousel/Carousel.tsx (hunk 2 above)
Comment (dmitry-yudakov) [reply packages/app/src/scenes/worldBuilder/pages/WorldBuilderCustomizePanel/components/Carousel/Carousel.tsx:19]: Couldn't figure out how to pass generic type to Carousel and use it like `<Carousel<MyItemType> items={items}>`

File: packages/app/src/api/repositories/assets3dRepository/assets3dRepository.api.endpoints.ts (hunk 1 above)
Comment (dmitry-yudakov) [reply packages/app/src/api/repositories/assets3dRepository/assets3dRepository.api.endpoints.ts]: then I'll rename this one... :)

Comment (Daboo1998) [reply packages/app/src/api/repositories/assets3dRepository/assets3dRepository.api.endpoints.ts]: I would rename it now, we don't use the word get and set in the filename.api.endpoints.ts file. Only use the word get and set in filename.api.types.ts and filename.api.ts

File: packages/app/src/scenes/worldBuilder/pages/WorldBuilderCustomizePanel/components/Carousel/Carousel.tsx (hunk 2 above)
Comment (Daboo1998) [reply packages/app/src/scenes/worldBuilder/pages/WorldBuilderCustomizePanel/components/Carousel/Carousel.tsx:19]: Wouldnt it infer the type by passing the items and then you dont need to use generic in tsx, but ofc still use generics in conponent definition/decleration.

File: packages/app/src/scenes/worldBuilder/pages/WorldBuilderCustomizePanel/components/Carousel/Carousel.tsx (hunk 3 above)
Comment (Daboo1998) [reply packages/app/src/scenes/worldBuilder/pages/WorldBuilderCustomizePanel/components/Carousel/Carousel.tsx:21]: No option of doing the same without three times the spread?

Comment (dmitry-yudakov) [reply packages/app/src/scenes/worldBuilder/pages/WorldBuilderCustomizePanel/components/Carousel/Carousel.tsx:21]: Lots of options of course but creating a temp array and then slicing the right part of it seems the simplest option :) The naming could be better of course...

Comment (Daboo1998) [reply packages/app/src/scenes/worldBuilder/pages/WorldBuilderCustomizePanel/components/Carousel/Carousel.tsx:21]: Yes then at least change the naming or add comments if not able to find a good and descriptive name

**Pull request merged**